File: strStr_BF.py
"""
Time complexity O(NM) where len of Haystack is N and len of needle is M 
 Brute force solution

"""


class Solution:
    def strStr(self, haystack: str, needle: str) -> int:
        if len(needle) ==0:
            return 0
        i=0
        while(i<len(haystack)):
            if(needle[0]==haystack[i]):
                j=0
                k=i
                while(k<len(haystack) and needle[j]==haystack[k]):
                    k+=1
                    j+=1
                    if(j==len(needle)):
                        return i
            i+=1
        return -1
        

File: strStr_BF.java
// 28. Implement strStr() - https://leetcode.com/problems/implement-strstr/
// Time Complexity :  O(MxN)
// Space Complexity : O(1)
// Did this code successfully run on Leetcode :yes
// Any problem you faced while coding this :no

// Your code here along with comments explaining your approach
// Brute Force

class Solution {
    public int strStr(String haystack, String needle) {
        if(needle == null || needle.length() == 0)
            return 0;
        if(haystack == null || haystack.length() == 0)
            return -1;
        int m = haystack.length();
        int n = needle.length();
        int i = 0;
        while(i < m){
            if(haystack.charAt(i) == needle.charAt(0)){
                int k = i + 1;
                int j = 1;
                while(k < m && j < n && haystack.charAt(k) == needle.charAt(j)){
                    k++;
                    j++;
                }
                if(j == n)
                    return i;
            }
            i++;
        }
        return -1;
    }
}